backend/utils/singleValueInMemoryCache.ts

```diff
@@ -0,0 +1,54 @@
+import moment from 'moment'
+import log from '../log'
+
+type StoredCachedData = {
+  expiryTime: moment.Moment
+  cachedData: any
+}
+
+let cachedValue: StoredCachedData = null
+
+class CacheHelper {
+  private currentDateTime: moment.Moment = moment()
+
+  private ttlInSeconds: number
+
+  constructor(ttlInSeconds: number) {
+    this.ttlInSeconds = ttlInSeconds
+  }
+
+  isExpired(cachedData: StoredCachedData): boolean {
+    return this.currentDateTime.isSameOrAfter(cachedData.expiryTime)
+  }
+
+  makeValueToCache(dataToCache: any): StoredCachedData {
+    return {
+      expiryTime: this.currentDateTime.add(this.ttlInSeconds, 'seconds'),
+      cachedData: dataToCache,
+    }
+  }
+}
+
+export const cacheFactory = (ttlInSeconds) => {
+  cachedValue = null
```

Comment:
So I think I get what is going on here - but... should this 'cachedValue' only be in scope here? I.e. not declared at the top of the file. That way every time we call the cacheFactory to create a cache we will get a different local value - which I think means it no longer has to be single use. Like we could call 
const contentfulCache = cacheFactory(30) and then
const anotherServiceCache = cacheFactory(60)?